src/cryptonote_core/uptime_proof.cpp

```diff
@@ -0,0 +1,111 @@
+#include "uptime_proof.h"
+#include "common/string_util.h"
+#include "version.h"
+
+extern "C"
+{
+#include <sodium/crypto_sign.h>
+}
+
+#undef OXEN_DEFAULT_LOG_CATEGORY
+#define OXEN_DEFAULT_LOG_CATEGORY "uptime_proof"
+
+//Default constructor for the uptime proof, will take the service node keys as a param and sign 
+uptime_proof::Proof::Proof(uint32_t public_ip, uint16_t storage_port, uint16_t storage_lmq_port, const std::array<uint16_t, 3> ss_version, uint16_t quorumnet_port, const std::array<uint16_t, 3> lokinet_version, const service_nodes::service_node_keys& keys)
```

Comment:
You should just wrap all the code in this file with `namespace uptime_proof {` rather than needing to namespace-qualify everything.